pypit/par/pypitpar.py

```diff
@@ -0,0 +1,2486 @@
+        # arguments
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        pars = dict([(k,values[k]) for k in args[1:]])      # "1:" to skip 'self'
+
+        # Initialize the other used specifications for this parameter
+        # set
+        defaults = dict.fromkeys(pars.keys())
+        options = dict.fromkeys(pars.keys())
+        dtypes = dict.fromkeys(pars.keys())
+        descr = dict.fromkeys(pars.keys())
+
+        # Fill out parameter specifications.  Only the values that are
+        # *not* None (i.e., the ones that are defined) need to be set
+
+        defaults['idsonly'] = False
+        dtypes['idsonly'] = bool
+        descr['idsonly'] = 'Only use the arc lines that have an identified wavelength to trace ' \
+                           'tilts'
+
+        defaults['method'] = 'spline'
+        options['method'] = TraceTiltsPar.valid_methods()
+        dtypes['method'] = str
+        descr['method'] = 'Method used to trace the tilt of the slit along an order.  ' \
+                          'Options are: {0}'.format(', '.join(options['method']))
+
+        # TODO: Need to add checks that check params against method
+        defaults['params'] = [ 1, 1, 0 ]
+        dtypes['params'] = [ int, list ]
+        descr['params'] = 'Parameters to use for the provided method.  TODO: Need more explanation'
+
+        defaults['order'] = 1
+        dtypes['order'] = int
+        descr['order'] = 'Order of the polynomial function to be used for the tilt of an ' \
+                         'individual arc line.  Must be 1 for eschelle data (ARMED pipeline).'
+
+        defaults['disporder'] = 1
+        dtypes['disporder'] = int
+        descr['disporder'] = 'Order of the polynomial function to be used to fit the tilts ' \
+                             'along the dispersion direction.  Only used by ARMED pipeline.'
+
+        # Instantiate the parameter set
+        super(TraceTiltsPar, self).__init__(list(pars.keys()),
+                                            values=list(pars.values()),
+                                            defaults=list(defaults.values()),
+                                            options=list(options.values()),
+                                            dtypes=list(dtypes.values()),
+                                            descr=list(descr.values()))
+        self.validate()
+
+    @classmethod
+    def from_dict(cls, cfg):
+        k = cfg.keys()
+        parkeys = [ 'idsonly', 'method', 'params', 'order', 'disporder' ]
+        kwargs = {}
+        for pk in parkeys:
+            kwargs[pk] = cfg[pk] if pk in k else None
+        return cls(**kwargs)
+
+    @staticmethod
+    def valid_methods():
+        """
+        Return the valid methods to use for tilt tracing.
+        """
+        return [ 'pca', 'spca', 'spline', 'interp', 'perp', 'zero' ]
+
+    def validate(self):
+        pass
+
+
+class TraceObjectsPar(ParSet):
```

Comment:
Object tracing is currently being overhauled  ;)

Reply:
Yep, will need to update these...